py/server/deephaven_internal/auto_completer/_completer.py

```diff
@@ -253,3 +258,36 @@ def do_hover(
             hoverstring += '\n---\n' + wrap_plaintext(raw_docstring)
 
         return hoverstring.strip()
+
+    def check_recursion_limit(self):
+        """
+        Tests for python+jedi+numpy versions that are susceptible to a RecursionError/segfault issue, and lowers
+        the recursion limit, warning if the limit is raised externally
+        """
+        if not sys.version.startswith('3.9.') and not sys.version.startswith('3.10.'):
+            return
+
+        if not self.__can_jedi:
+            return
+
+        # numpy is a required dependency, tested for jedi above
+        if not version('numpy').startswith('2.0') or not version('jedi') == '0.19.1':
+            return
+
+        from . import MAX_RECURSION_LIMIT
```

Comment:
Is there a reason that `MAX_RECURSION_LIMIT` is not included in this file instead of in the root?  Do you expect it to be used elsewhere?

Reply:
This way it can easily be set by a user if they want to fine tune the limit further. Given that we don't understand why the issue is happening to begin with, it might help to let us tweak it at runtime without a new release.